File: java/get-complaint-types.java
import okhttp3.*;

import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        String apiUrl = "<REPLACE_WITH_PROD_URL>/api/v1/complaint-types";
        String token = "<REPLACE_WITH_YOUR_API_KEY>";

        OkHttpClient client = new OkHttpClient().newBuilder().build();

        Request request = new Request.Builder()
                .url(apiUrl)
                .method("GET", null)
                .addHeader("Authorization", "Bearer " + token)
                .addHeader("Content-Type", "application/json")
                .build();

        Response response = client.newCall(request).execute();
        String responseBody = response.body().string();

        if (response.isSuccessful()) {
            System.out.println(responseBody);
        } else {
            System.out.println("Erreur lors de la récupération des types de réclamations : " + responseBody);
        }
    }
}

File: csharp/get-complaint-types.cs
using RestSharp;
using System;

class Program
{
    static void Main(string[] args)
    {
        string apiUrl = "<REPLACE_WITH_PROD_URL>/api/v1/complaint-types";
        string token = "<REPLACE_WITH_YOUR_API_KEY>";

        var client = new RestClient(apiUrl);
        client.Timeout = -1;

        var request = new RestRequest(Method.GET);
        request.AddHeader("Authorization", $"Bearer {token}");
        request.AddHeader("Content-Type", "application/json");

        IRestResponse response = client.Execute(request);

        if (response.IsSuccessful)
        {
            Console.WriteLine(response.Content);
        }
        else
        {
            Console.WriteLine($"Erreur lors de la récupération des types de réclamations : {response.ErrorMessage}");
        }
    }
}
